dlt/common/destination/reference.py

```diff
@@ -449,12 +469,14 @@ def client_class(self) -> Type[TDestinationClient]:
 
     def configuration(self, initial_config: TDestinationConfig) -> TDestinationConfig:
         """Get a fully resolved destination config from the initial config"""
-        return resolve_configuration(
+        config = resolve_configuration(
             initial_config,
             sections=(known_sections.DESTINATION, self.name),
             # Already populated values will supersede resolved env config
             explicit_value=self.config_params,
         )
+        config.name = self.name
```

Comment:
why? we pass the `name` to kwargs in init. are you protecting against someone overwriting the name from config? then place `name` as `Final[str]`

Reply:
this was the place to set name or type if name was not set, but I put that in an onresolved method now.